modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/authenticator/OAuthTokenValidationStubFactory.java

```diff
@@ -0,0 +1,209 @@
+     * @return An instance of properly configured MultiThreadedHttpConnectionManager
+     */
+    private HttpConnectionManager createConnectionManager(Properties properties) {
+        HttpConnectionManagerParams params = new HttpConnectionManagerParams();
+        if (properties == null || properties.isEmpty()) {
+            throw new IllegalArgumentException("Parameters required to initialize HttpClient instances " +
+                    "associated with OAuth token validation service stub are not provided");
+        }
+        String maxConnectionsPerHostParam = properties.getProperty("MaxConnectionsPerHost");
+        if (maxConnectionsPerHostParam == null || maxConnectionsPerHostParam.isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("MaxConnectionsPerHost parameter is not explicitly defined. Therefore, the default, " +
+                        "which is 2, will be used");
+            }
+        } else {
+            params.setDefaultMaxConnectionsPerHost(Integer.parseInt(maxConnectionsPerHostParam));
+        }
+
+        String maxTotalConnectionsParam = properties.getProperty("MaxTotalConnections");
+        if (maxTotalConnectionsParam == null || maxTotalConnectionsParam.isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("MaxTotalConnections parameter is not explicitly defined. Therefore, the default, " +
+                        "which is 10, will be used");
+            }
+        } else {
+            params.setMaxTotalConnections(Integer.parseInt(maxTotalConnectionsParam));
+        }
+        HttpConnectionManager connectionManager = new MultiThreadedHttpConnectionManager();
+        connectionManager.setParams(params);
+        return connectionManager;
+    }
+
+    /**
+     * Creates an instance of PoolingHttpClientConnectionManager using HttpClient 4.x APIs
+     *
+     * @param properties Properties to configure PoolingHttpClientConnectionManager
+     * @return An instance of properly configured PoolingHttpClientConnectionManager
+     */
+    private HttpClientConnectionManager createClientConnectionManager(Properties properties) {
+        PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+        if (properties != null) {
+            String maxConnectionsPerHostParam = properties.getProperty("MaxConnectionsPerHost");
+            if (maxConnectionsPerHostParam == null || maxConnectionsPerHostParam.isEmpty()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("MaxConnectionsPerHost parameter is not explicitly defined. Therefore, the default, " +
+                            "which is 2, will be used");
+                }
+            } else {
+                connectionManager.setDefaultMaxPerRoute(Integer.parseInt(maxConnectionsPerHostParam));
+            }
+
+            String maxTotalConnectionsParam = properties.getProperty("MaxTotalConnections");
```

Comment:
use constants to get MaxTotalConnections from https://github.com/wso2/product-mdm/pull/156